crates/wash-lib/src/cli/mod.rs

```diff
@@ -269,7 +270,31 @@ pub fn extract_keypair(
         }
     } else if let Some(module) = module_path {
         // No seed value provided, attempting to source from provided or default directory
-        let key_dir = KeyDir::new(determine_directory(directory)?)?;
+        let key_dir = {
+            match keygen_type {
+                KeyPairType::Module => {
+                    let default = || -> Result<KeyDir> { KeyDir::new(determine_directory(None)?) };
+
+                    match directory {
+                        Some(d) => KeyDir::new(d),
+                        None => {
+                            if let Ok(project_config) = get_config(None, Some(true)) {
+                                match project_config.project_type {
+                                    TypeConfig::Actor(ref actor_config) => {
+                                        KeyDir::new(&actor_config.key_directory)
+                                    }
+                                    _ => default(),
+                                }
+                            } else {
+                                default()
+                            }
+                        }
+                    }
+                }
+                _ => KeyDir::new(determine_directory(directory)?),
+            }
```

Comment:
This statement may be more human-readable if the output was the directory that needed to be wrapped in `KeyDir::new()`, if that makes sense. 

My main issue with this setup is the function `default`, which is a nice use of a closure but is assigning to a name that seems like a reserved keyword.

Reply:
i updated the logic, let me know if that works